client/src/frontend/components/resource-player/Video/index.js

```diff
@@ -104,23 +104,49 @@ class ResourcePlayerVideo extends Component {
     this.props.dispatch(notificationActions.addNotification(notification));
   };
 
+  trackRef = el => {
+    if (!el) return;
+
+    el.addEventListener("load", () => {
+      if (!el.track?.cues) return;
+      Array.from(el.track.cues).forEach(c => {
+        // eslint-disable-next-line no-param-reassign
+        c.line = -3;
+      });
+    });
+  };
+
   renderFileVideo() {
     if (!this.state.inBrowser) return null;
 
     const {
       variantPosterStyles,
-      attachmentStyles
+      attachmentStyles,
+      captionsTrackUrl
     } = this.props.resource.attributes;
 
+    // Use the proxy if running over ports in dev to avoid CORS error
+    const captionsSrc =
+      process.env.NODE_ENV === "development"
+        ? captionsTrackUrl.replace("3020", "3010")
+        : captionsTrackUrl;
```

Comment:
I know it's just for dev, but string replacing like this seems liable to alter some urls that we don't want to alter. Could you use the URL interface here and change the `port` property instead?